docker/Dockerfile

```diff
@@ -1,72 +1,45 @@
-# use debian as base image
-FROM debian:stretch
+# Use mambaforge as the base image
+# Builds in ~ 4.25 min and is ~ 2.6 GB on a windows laptop
+FROM mambaorg/micromamba:0.15.3
 
 # Label image
 LABEL \
     "Description"="Container for open source time series InSAR processing with Mintpy" \
     "Github Source"="https://github.com/insarlab/MintPy/" \
     "Installation"="https://github.com/insarlab/MintPy/blob/main/docs/installation.md" \
-    "Dockerfile Author"="Andre Theron" \
-    "Email"="andretheronsa@gmail.com"
+    "Dockerfile Author"="Forrest Williams" \
+    "Email"="forrestfwilliams@icloud.com"
 
-# Install useful programs
+# Install git and vim
+USER root
 RUN apt-get update && \
-    apt-get install -y \
-    git \
-    python3 \
-    python3-pip \
-    wget \ 
-    vim \
-    zip \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
+    apt-get install -y git vim && \
+    rm -rf /var/lib/apt/lists/*
 
-# Change default shell from dash to bash permanently
-SHELL ["/bin/bash", "-c"]
+# Clone MintPy and PyAPS repositories
+USER micromamba
+WORKDIR /home/micromamba
 
-# Prepare build time environment for conda and mintpy
-ARG MINTPY_HOME=/home/python/MintPy
-ARG PYTHON3DIR=/home/python/miniconda3
-ARG PYAPS_HOME=/home/python/PyAPS
-ARG WORK_DIR=/home/work/
+RUN mkdir /home/micromamba/tools && \
+    git clone https://github.com/insarlab/MintPy.git /home/micromamba/tools/MintPy && \
+    git clone https://github.com/yunjunz/PyAPS.git /home/micromamba/tools/PyAPS
+
+# Prepare environment for MintPy
+ARG MINTPY_HOME=/home/micromamba/tools/MintPy
+ARG PYAPS_HOME=/home/micromamba/tools/PyAPS
+ARG PYTHON3DIR=/opt/conda
 
-# Prepare container environment
 ENV MINTPY_HOME=${MINTPY_HOME}
-ENV PYTHON3DIR=${PYTHON3DIR}
 ENV PYAPS_HOME=${PYAPS_HOME}
 ENV PATH=${MINTPY_HOME}/mintpy:${PYTHON3DIR}/bin:${PATH}
-ENV PYTHONPATH=${PYTHONPATH}:${MINTPY_HOME}:${PYAPS_HOME}
-ENV PROJ_LIB=${PYTHON3DIR}/share/proj
-
-# Set workdir
-WORKDIR /home/python
-
-# Pull and config mintpy
-RUN git clone https://github.com/insarlab/MintPy.git
-
-# Pull and config PyAPS
-RUN git clone https://github.com/yunjunz/PyAPS.git
-
-# Install miniconda
-RUN wget https://repo.continuum.io/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh && \
-    chmod +x ./Miniconda3-4.5.4-Linux-x86_64.sh  && \
-    ./Miniconda3-4.5.4-Linux-x86_64.sh -b -p ${PYTHON3DIR} && \
-    rm ./Miniconda3-4.5.4-Linux-x86_64.sh
-
-# Install dependencies
-RUN ${PYTHON3DIR}/bin/conda config --add channels conda-forge && \
-    ${PYTHON3DIR}/bin/conda install --yes --file ${MINTPY_HOME}/docs/requirements.txt
-
-# Install pykml
-RUN ${PYTHON3DIR}/bin/pip install git+https://github.com/yunjunz/pykml.git
+ENV PYTHONPATH=${MINTPY_HOME}:${PYAPS_HOME}
```

Comment:
Since `MINTPY_HOME` and `PYAPS_HOME` are only used in the building and not needed afterward. We could use the `ARG` only, without `ENV` for them.